internal/coreinternal/golden/normalize_timestamps.go

```diff
@@ -0,0 +1,198 @@
+			if attrHash == pdatautil.MapHash(sdps.At(j).Attributes()) {
+				ts = append(ts, sdps.At(j).Timestamp())
+			}
+		}
+
+		normalizedTs := normalizeTimestampSlice(ts)
+		for k := 0; k < sdps.Len(); k++ {
+			if pdatautil.MapHash(sdps.At(k).Attributes()) == attrHash {
+				for key, value := range normalizedTs {
+					if sdps.At(k).Timestamp() == value {
+						sdps.At(k).SetTimestamp(key)
+					}
+				}
+			}
+		}
+		hashCache[attrHash] = true
+	}
+}
+
+func normExponentialHistogramDataPointSlice(ehdps pmetric.ExponentialHistogramDataPointSlice) {
+	hashCache := make(map[[16]byte]bool)
+	for i := 0; i < ehdps.Len(); i++ {
+		attrHash := pdatautil.MapHash(ehdps.At(i).Attributes())
+		if hashCache[attrHash] {
+			continue
+		}
+		ts := []pcommon.Timestamp{ehdps.At(i).Timestamp()}
+		for j := i + 1; j < ehdps.Len(); j++ {
+			if attrHash == pdatautil.MapHash(ehdps.At(j).Attributes()) {
+				ts = append(ts, ehdps.At(j).Timestamp())
+			}
+		}
+
+		normalizedTs := normalizeTimestampSlice(ts)
+		for k := 0; k < ehdps.Len(); k++ {
+			if pdatautil.MapHash(ehdps.At(k).Attributes()) == attrHash {
+				for key, value := range normalizedTs {
+					if ehdps.At(k).Timestamp() == value {
+						ehdps.At(k).SetTimestamp(key)
+					}
+				}
+			}
+		}
+		hashCache[attrHash] = true
+	}
+}
+
+func normHistogramDataPointSlice(hdps pmetric.HistogramDataPointSlice) {
+	hashCache := make(map[[16]byte]bool)
+	for i := 0; i < hdps.Len(); i++ {
+		attrHash := pdatautil.MapHash(hdps.At(i).Attributes())
+		if hashCache[attrHash] {
+			continue
+		}
+		ts := []pcommon.Timestamp{hdps.At(i).Timestamp()}
+		for j := i + 1; j < hdps.Len(); j++ {
+			if attrHash == pdatautil.MapHash(hdps.At(j).Attributes()) {
+				ts = append(ts, hdps.At(j).Timestamp())
+			}
+		}
+
+		normalizedTs := normalizeTimestampSlice(ts)
+		for k := 0; k < hdps.Len(); k++ {
+			if pdatautil.MapHash(hdps.At(k).Attributes()) == attrHash {
+				for key, value := range normalizedTs {
+					if hdps.At(k).Timestamp() == value {
+						hdps.At(k).SetTimestamp(key)
+					}
+				}
+			}
+		}
+		hashCache[attrHash] = true
+	}
+}
+
+// sorts a slice of pcommon.Timestamp and returns a normalized slice of pcommon.Timestamp
+// normalization entails setting nonunique subsequent timestamps to the same value while incrementing unique timestamps by a set value of 1,000,000 ns
+func normalizeTimestampSlice(ts []pcommon.Timestamp) map[pcommon.Timestamp]pcommon.Timestamp {
+	normalizedTs := make(map[pcommon.Timestamp]pcommon.Timestamp)
+	tempTs := make([]pcommon.Timestamp, len(ts))
+	copy(tempTs, ts)
+	sort.Slice(tempTs, func(i, j int) bool {
+		return func(t1, t2 pcommon.Timestamp) int {
+			if t1 < t2 {
+				return -1
+			} else if t1 > t2 {
+				return 1
+			}
+			return 0
+		}(tempTs[i], tempTs[j]) < 0
+	})
+
+	rc := time.Nanosecond * 0
+	prevTime := pcommon.NewTimestampFromTime(time.Unix(0, 0))
+
+	for i := range ts {
+		if tempTs[i] != prevTime {
+			rc += 1000000 * time.Nanosecond
```

Comment:
Let's move this into a function so we can isolate the logic and change it later if necessary. It shouldn't be dependent on knowing the previous data point's time - just make it functional. Roughly: 
```go
func normalTime(timeSeriesIndex int) pcommon.Timestamp {
    return pcommon.NewTimestampFromTime(time.Unix(0, 0).Add(i * time.Second))
}
```